app/post.go

```diff
@@ -545,6 +545,10 @@ func (a *App) SendEphemeralPost(c request.CTX, userID string, post *model.Post)
 	post.GenerateActionIds()
 	message := model.NewWebSocketEvent(model.WebsocketEventEphemeralMessage, "", post.ChannelId, userID, nil, "")
 	post = a.PreparePostForClientWithEmbedsAndImages(c, post, true, false, true)
+	post, appErr := a.SanitizePostMetadataForUser(c, post, userID)
+	if appErr != nil {
+		mlog.Error("Failed to sanitize post metadata for user", mlog.String("user_id", userID), mlog.Err(appErr))
```

Comment:
If we weren't able to sanitize the data, we should `return` from here instead of continuing and risking leaking information. We're doing this in other places as well.

Reply:
In fact if we have to return from here as `post` can be nil in case of error and we'll end up with nil pointer deference error.

Reply:
Yes, that's a good catch. I thought we could avoid changing the method's signature and not handling the error.

We could do something like this to improve it but it could potentially leak information.
```go
	sanitizedPost, appErr := a.SanitizePostMetadataForUser(c, post, userID)
	if appErr != nil {
		mlog.Error("Failed to sanitize post metadata for user", mlog.String("user_id", userID), mlog.Err(appErr))
	}
	if sanitizedPost != nil {
		post = sanitizedPost
	}
```

But it looks like not returning an error, in this case, could lead to more issues.